### PR TITLE
Replace bind1st and bind2nd by bind

### LINKED ISSUE #481: c++17 support?
The clang folks have implemented c++17 a bit more aggressively than gcc: some members that were deprecated in c++17 are removed, like `std::bind1st`, `std::bind2nd` as used in e.g. [` vigra/include/vigra/numpy_array_traits.hxx `](https://github.com/ukoethe/vigra/blob/fb427440da8c42f96e14ebb60f7f22bdf0b7b1b2/include/vigra/numpy_array_traits.hxx#L765).

gcc [seems to be keeping those](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91383) for the time being.

For projects that link against vigra this poses a problem: on osx those projects cannot be compiled using `c++17`. Are there any plans to add support for this? Or, are there plans to support standards older than `c++11`? Otherwise one can use lambdas instead...

### PR DESCRIPTION
**Pull request by constantinpape**
Closes #481

### CODE REVIEW

**Comment by hmeine** [comment]
Great; I have not been monitoring my GitHub notifications, so I missed this, but I am glad you took this on!

**Comment by constantinpape** [comment]
Thanks for merging this @hmeine!
Do you think there is any chance to get a new vigra release at some point?
The conda-forge recipe for vigra is full of patches for the changes in this PR and others that are merged long ago and it would be great to clean this up at some point. See https://github.com/conda-forge/vigra-feedstock/blob/master/recipe/meta.yaml.

**Comment by hmeine** [comment]
Good question. I would not want to do that without talking to @ukoethe. But I think we should find a way to define a streamlined release process and get a new official version out with all the fixes since the latest release – would be a big pity if we couldn't achieve that!

**Comment by constantinpape** [comment]
Thanks @hmeine! Do you want to contact @ukoethe about this? (I think he also doesn't check github notifications regularly right now). Or I could also write him a mail about this.

**Comment by hmeine** [comment]
Good idea, then maybe you write an email to him with me in CC (MEVIS, GMX or Uni Bremen addresses would all work).

File: test/imagehierarchy/single_band_image_test.hxx [hunk 1]
```diff
@@ -29,11 +29,11 @@ public:
         */
         image1->init(data[5]); 
         should((*image1_->begin()) != static_cast<typename Image::PixelType> (data[5]));
-        should(image1->end() == std::find_if(image1->begin(), image1->end(), std::bind2nd(Pixels_not_equal_to<value_type>(), data[5])));
+        should(image1->end() == std::find_if(image1->begin(), image1->end(), std::bind(Pixels_not_equal_to<value_type>(), data[5], std::placeholders::_2)));
         
         image1_->init(data[6]);
-        should(image1->end() == std::find_if(image1->begin(), image1->end(), std::bind2nd(Pixels_not_equal_to<value_type>(), data[5])));
-        should(image1_->end() == std::find_if(image1_->begin(), image1_->end(), std::bind2nd(Pixels_not_equal_to<value_type>(), data[6])));
+        should(image1->end() == std::find_if(image1->begin(), image1->end(), std::bind(Pixels_not_equal_to<value_type>(), data[5])));
```
Comment (hmaarrfk) [inline test/imagehierarchy/single_band_image_test.hxx:35]: hmm, this is a typo.

**Pull request merged**